lib/membrane_sdl/player.ex

```diff
@@ -15,72 +16,58 @@ defmodule Membrane.SDL.Player do
   # The measured latency needed to show a frame on a screen.
   @latency 20 |> Time.milliseconds()
 
-  def_input_pad :input, caps: RawVideo, demand_unit: :buffers
+  def_input_pad :input, accepted_format: RawVideo, demand_unit: :buffers
 
   @impl true
-  def handle_init(_options) do
+  def handle_init(_options, _ctx) do
     state = %{cnode: nil, timer_started?: false}
-    {{:ok, latency: @latency}, state}
+    {[latency: @latency], state}
   end
 
   @impl true
-  def handle_stopped_to_prepared(_ctx, state) do
+  def handle_setup(_ctx, state) do
     {:ok, cnode} = CNode.start_link(:player)
-    {:ok, %{state | cnode: cnode}}
+
+    # TODO CNODE is dead after the pipeline is stopped, investigate why
+    # Membrane.ResourceGuard.register(
+    #   ctx.resource_guard,
+    #   fn -> CNode.stop(cnode) end
+    # )
```

Comment:
It's probably because the C node automatically dies when its parent process dies

Reply:
So `ResourceGuard` can be omitted, can't it?

Reply:
yes